### PR TITLE
doc improvements

### PR DESCRIPTION
**Pull request by BurnzZ**
Ignore the failing tests, it's because of https://github.com/scrapinghub/web-poet/pull/124 which was included in the latest **web-poet** release.

It's fixed in https://github.com/scrapinghub/scrapy-poet/pull/120.

### CODE REVIEW

File: docs/intro/basic-tutorial.rst [hunk 1]
```diff
@@ -418,9 +418,9 @@ be done by configuring ``SCRAPY_POET_RULES`` into ``settings.py``:
 
 .. code-block:: python
 
-    "SCRAPY_POET_RULES": [
-        ("toscrape.com", BTSBookListPage, BookListPage),
-        ("toscrape.com", BTSBookPage, BookPage)
+    SCRAPY_POET_RULES = [
+        ApplyRule("toscrape.com", BTSBookListPage, BookListPage),
+        ApplyRule("toscrape.com", BTSBookPage, BookPage)
```
Comment (BurnzZ) [inline docs/intro/basic-tutorial.rst:423]: Missed in #103

File: scrapy_poet/downloadermiddlewares.py [hunk 2]
```diff
@@ -116,17 +117,13 @@ def _skip_dependency_creation(self, request: Request, spider: Spider) -> bool:
...
+        for the required Page Objects found in the callback signature.
 
-        In other words, this method instantiates all ``Injectable``
-        subclasses declared as request callback arguments and
-        any other parameter with a ``PageObjectInputProvider`` configured for
-        its type.
-
-        If there's a collision between an already set ``cb_kwargs``
-        and an injectable attribute,
-        the user-defined ``cb_kwargs`` takes precedence.
```
Comment (BurnzZ) [inline scrapy_poet/downloadermiddlewares.py:129]: Missed in #106

File: scrapy_poet/page_input_providers.py [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 that the Page Objects uses to get external data (e.g. the HTML). That's why we
-have created a repository of ``PageObjectInputProviders``.
+have created a colletion of Page Object Input Providers.
 
-The current module implements a ``PageObjectInputProviders`` for
-:class:`web_poet.page_inputs.HttpResponse`, which is in charge of providing the response
-HTML from Scrapy. You could also implement different providers in order to
-acquire data from multiple external sources, for example,
-Splash or Auto Extract API.
+The current module implements a Page Input Providers for
```
Comment (Gallaecio) [inline scrapy_poet/page_input_providers.py:5]: In plain English, maybe it should be singular, “provider”.

File: scrapy_poet/page_input_providers.py [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+:class:`web_poet.HttpResponse <web_poet.page_inputs.http.HttpResponse>`, which
+is in charge of providing the response HTML from Scrapy. You could also implement
+different providers in order to acquire data from multiple external sources,
+for example, Splash or Auto Extract API.
```
Comment (Gallaecio) [inline scrapy_poet/page_input_providers.py:9]: Unrelated, but I wonder if we should take the chance to:

```suggestion
for example, from scrapy-playwright or from an API for automatic extraction.
```

**Review by Gallaecio — APPROVED** [review]

File: scrapy_poet/downloadermiddlewares.py (hunk 2 above)
Comment (kmike) [reply scrapy_poet/downloadermiddlewares.py:129]: nice catch
